dpbench/console/dpbench_cli.py

```diff
@@ -0,0 +1,177 @@
+
+
+class Namespace(argparse.Namespace):
+    benchmarks: set[str]
+    implementations: set[str]
+    all_implementations: bool
+    preset: str
+    skip_validation: bool
+    dpbench: bool
+    npbench: bool
+    polybench: bool
+    print_results: bool
+    validate: bool
+    run_id: int | None
+    last_run: bool
+    results_db: str
+    save: bool
+    repeat: int
+    timeout: float
+    precision: str | None
+    program: str
+
+
+class CommaSeparateStringAction(argparse.Action):
+    def __call__(self, _, namespace, values, __):
+        setattr(namespace, self.dest, set(values.split(",")))
+
+
+def parse_args() -> Namespace:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-b",
+        "--benchmarks",
+        type=str,
+        action=CommaSeparateStringAction,
+        nargs="?",
+        default={},
+    )
+    parser.add_argument(
+        "-i",
+        "--implementations",
+        type=str,
+        action=CommaSeparateStringAction,
+        nargs="?",
+        default={"python", "numpy"},
+    )
+    parser.add_argument(
+        "-a",
+        "--all-implementations",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
+    # TODO: set proper version
+    parser.add_argument("--version", action="version", version="%(prog)s 2.0")
+    parser.add_argument("-r", "--run-id", type=int, nargs="?", default=None)
+    parser.add_argument(
+        "--last-run", action=argparse.BooleanOptionalAction, default=False
+    )
+    parser.add_argument(
+        "--results-db", type=str, nargs="?", default="results.db"
+    )
+
+    subparsers = parser.add_subparsers(dest="program")
+
+    run_parser = subparsers.add_parser("run")
+
+    run_parser.add_argument(
+        "-p",
+        "--preset",
+        choices=["S", "M", "L"],
+        type=str,
+        nargs="?",
+        default="S",
+    )
+    run_parser.add_argument(
+        "-s",
+        "--skip-validation",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
+    run_parser.add_argument(
+        "--dpbench", action=argparse.BooleanOptionalAction, default=True
+    )
+    run_parser.add_argument(
+        "--npbench", action=argparse.BooleanOptionalAction, default=False
+    )
+    run_parser.add_argument(
+        "--polybench", action=argparse.BooleanOptionalAction, default=False
+    )
+    run_parser.add_argument("-r", "--repeat", type=int, nargs="?", default=10)
+    run_parser.add_argument(
+        "-t", "--timeout", type=float, nargs="?", default=200.0
+    )
+    run_parser.add_argument(
+        "--precision",
+        choices=["single", "double"],
+        type=str,
+        nargs="?",
+        default=None,
+    )
+    run_parser.add_argument(
+        "--print-results", action=argparse.BooleanOptionalAction, default=True
+    )
+    run_parser.add_argument(
+        "--save", action=argparse.BooleanOptionalAction, default=True
+    )
+
```

Comment:
We need one more argument:

```python
run_parser.add_argument(
        "--device",
        choices=["cpu", "gpu"],
        type=str,
        nargs="?",
        default=None,
    )
```
If this argument is provided, it should override the `sycl_device` argument in `cfg`.

Reply:
Lets add it in the following RP. Don't want to blow this one.